errcheck/errcheck.go

```diff
@@ -124,6 +104,41 @@ func (e byName) Less(i, j int) bool {
 	return ei.Line < ej.Line
 }
 
+// Append appends errors to e. Append does not do any duplicate checking.
+func (r *Result) Append(other *Result) {
+	r.UncheckedErrors = append(r.UncheckedErrors, other.UncheckedErrors...)
+}
+
+// Returns the unique errors that have been accumulated. Duplicates may occur
+// when a file containing an unchecked error belongs to > 1 package.
+//
+// This function works in place, but returns the result anyways.
+func (r *Result) Unique() *Result {
+	sort.Sort(byName{r})
```

Comment:
This looks like the only place sort interface implementation is used. I would suggest to just sort the errors list directly and remove sort interface from result for simplicity. Probably also good to have this not mutate the result and sort a copy of the list